### PR TITLE
Adding BERTScore metric

### LINKED ISSUE #365: BERTScore Implementation
## 🚀 Feature
BertScore Implementation for TextGeneration in NLP

### Motivation

BERTScore computes a similarity score for each token in the candidate sentence with each token in the reference sentence. However, instead of exact matches, we compute token similarity using contextual embeddings. 

### Pitch

need an api like `bert_score(translate_corpus, reference_corpus, ...)`

Related Paper: https://arxiv.org/abs/1904.09675

A Really Good Explanation: https://wiki.math.uwaterloo.ca/statwiki/index.php?title=BERTScore:_Evaluating_Text_Generation_with_BERT

### Alternatives

This is an implementation: https://github.com/Tiiiger/bert_score

### PR DESCRIPTION
**Pull request by gagan3012**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
This PR adds the BERTScore metric from its implementation here: https://github.com/Tiiiger/bert_score
Fixes #365 (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/functional/text/bertscore.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        - :param: `idf` (bool or dict): use idf weighting, can also be a precomputed idf_dict
+        - :param: `device` (str): on which the contextual embedding model will be allocated on.
+                  If this argument is None, the model lives on cuda:0 if cuda is available.
+        - :param: `nthreads` (int): number of threads
+        - :param: `batch_size` (int): bert score processing batch size
+        - :param: `lang` (str): language of the sentences; has to specify
+                  at least one of `model_type` or `lang`. `lang` needs to be
+                  specified when `rescale_with_baseline` is True.
+        - :param: `rescale_with_baseline` (bool): rescale bertscore with pre-computed baseline
+        - :param: `baseline_path` (str): customized baseline file
```
Comment (Borda) [inline torchmetrics/functional/text/bertscore.py:60]: pls use Napoleon style
+ types are already in func signature

File: torchmetrics/functional/text/bertscore.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+# limitations under the License.
+from typing import Dict, List
+
+from torchmetrics.utilities.imports import _BERTSCORE_AVAILABLE
+
+if _BERTSCORE_AVAILABLE:
+    import bert_score
+
+
+def bertscore(
```
Comment (Borda) [inline torchmetrics/functional/text/bertscore.py:22]:
```suggestion
def bert_score(
```

File: torchmetrics/functional/text/bertscore.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
+# Copyright The PyTorch Lightning team.
```
Comment (Borda) [inline torchmetrics/functional/text/bertscore.py:1]: lets name the file just `bert.py`

File: torchmetrics/functional/text/bertscore.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            nthreads=nthreads,
+            all_layers=all_layers,
+            idf=idf,
+            device=device,
+            lang=lang,
+            rescale_with_baseline=rescale_with_baseline,
+            baseline_path=baseline_path,
+        )
+
+    (P, R, F) = cached_bertscorer.score(
```
Comment (Borda) [inline torchmetrics/functional/text/bertscore.py:116]:
```suggestion
    P, R, F = cached_bertscorer.score(
```

File: torchmetrics/functional/text/bertscore.py (hunk 2 above)
Comment (gagan3012) [reply torchmetrics/functional/text/bertscore.py:22]: The supporting package is also called `bert_score`

Comment (Borda) [reply torchmetrics/functional/text/bertscore.py]: Yes, but you can import as.... Or even better, you shall import only function you are going to use =)

Comment (gagan3012) [reply torchmetrics/functional/text/bertscore.py]: Found a way thanks

File: docs/source/references/functional.rst [hunk 5]
```diff
@@ -382,3 +382,8 @@ wer [func]
 
 .. autofunction:: torchmetrics.functional.wer
     :noindex:
+
+bert_score [func]
```
Comment (SkafteNicki) [inline docs/source/references/functional.rst:386]: could you move this to be the first metric in the text section (alphabetical order)?

File: docs/source/references/modules.rst [hunk 6]
```diff
@@ -543,6 +543,11 @@ WER
 .. autoclass:: torchmetrics.WER
     :noindex:
 
+BERTScore
```
Comment (SkafteNicki) [inline docs/source/references/modules.rst:546]: could you move this to be the first metric in the text section (alphabetical order)?

File: torchmetrics/functional/text/bert.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+                      the *best* score among all references.
+
+    Example:
+        >>> predictions = ["hello there", "general kenobi"]
+        >>> references = ["hello there", "general kenobi"]
+        >>> results = bert_score(predictions=predictions, references=references, lang="en")
+        >>> print([round(v, 2) for v in results["f1"]])
+        [1.0, 1.0]
+    """
+    if model_type is None:
```
Comment (SkafteNicki) [inline torchmetrics/functional/text/bert.py:76]:
```suggestion
    if not _BERTSCORE_AVAILABLE:
        raise ValueError("bert_score metric requires that bert-score package is installed."
                          " Either install with `pip install bert-score` or `pip install torchmetrics[text]`")

    if model_type is None:
```

File: torchmetrics/text/bert.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        >>> predictions = ["hello there", "general kenobi"]
+        >>> references = ["hello there", "general kenobi"]
+        >>> results = bert_score(predictions=predictions, references=references, lang="en")
+        >>> print([round(v, 2) for v in results["f1"]])
+        [1.0, 1.0]
+    """
+
+    def __init__(
+        self,
+        model_type: str = None,
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:65]:
```suggestion
        model_type: Optional[str] = None,
```

File: torchmetrics/text/bert.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+class BERTScore(Metric):
+    """BERTScore leverages the pre-trained contextual embeddings from BERT and matches words in candidate and
+    reference sentences by cosine similarity. It has been shown to correlate with human judgment on sentence-level
+    and system-level evaluation. Moreover, BERTScore computes precision, recall, and F1 measure, which can be
+    useful for evaluating different language generation tasks.
+
+    Args:
+        - :param: `cands` (list of str): candidate sentences
+        - :param: `refs` (list of str or list of list of str): reference sentences
+        - :param: `model_type` (str): bert specification, default using the suggested
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:29]: please remove the :param: from all 
```suggestion
        - `model_type` (str): bert specification, default using the suggested
```

File: torchmetrics/text/bert.py [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+        - :param: `rescale_with_baseline` (bool): rescale bertscore with pre-computed baseline
+        - :param: `baseline_path` (str): customized baseline file
+        - :param: `use_fast_tokenizer` (bool): `use_fast` parameter passed to HF tokenizer
+
+    Returns:
+        - :param: `(P, R, F)`: each is of shape (N); N = number of input
+                      candidate reference pairs. if returning hashcode, the
+                      output will be ((P, R, F), hashcode). If a candidate have
+                      multiple references, the returned score of this candidate is
+                      the *best* score among all references.
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:53]: move this to the compute method

File: torchmetrics/text/bert.py [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+        >>> references = ["hello there", "general kenobi"]
+        >>> results = bert_score(predictions=predictions, references=references, lang="en")
+        >>> print([round(v, 2) for v in results["f1"]])
+        [1.0, 1.0]
+    """
+
+    def __init__(
+        self,
+        model_type: str = None,
+        num_layers: int = None,
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:66]:
```suggestion
        num_layers: Optional[int] = None,
```

File: torchmetrics/text/bert.py [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+        verbose: bool = False,
+        idf: bool = False,
+        device: str = None,
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:69]:
```suggestion
        device: Optional[str] = None,
```

File: torchmetrics/text/bert.py [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+        batch_size: int = 64,
+        nthreads: int = 4,
+        all_layers: bool = False,
+        lang: str = None,
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:73]:
```suggestion
        lang: Optional[str] = None,
```

File: torchmetrics/text/bert.py [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+        rescale_with_baseline: bool = False,
+        baseline_path: str = None,
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:75]:
```suggestion
        baseline_path: Optional[str] = None,
```

File: tests/text/test_bertscore.py [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+@pytest.mark.parametrize(
+    "preds,refs",
+    [(preds, refs)],
+)
+@pytest.mark.skipif(not _BERTSCORE_AVAILABLE, reason="test requires bert_score")
+def test_score_fn(preds, refs):
+    Score = bert_score(preds, refs, model_type="roberta-large", num_layers=17, idf=False, batch_size=3)
+    assertTensorsAlmostEqual(
+        expected=Score["precision"], actual=[0.9843302369117737, 0.9832239747047424, 0.9120386242866516]
```
Comment (SkafteNicki) [inline tests/text/test_bertscore.py:40]: where does the values comes from?

**Review by SkafteNicki — COMMENTED** [review]
Some initial comments, PR looks great already

File: tests/text/test_bertscore.py [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def assertTensorsAlmostEqual(expected, actual, decimal=5):
+    """
+    Test tensors are almost equal (EPS = 1e-5 by default)
+    """
+    np.testing.assert_almost_equal(expected, actual, decimal=decimal)
+
+
+preds = [
```
[the remark this answers is not in the document]
Comment (gagan3012) [reply tests/text/test_bertscore.py:16]: These are from the bert-score repo

File: tests/text/test_bertscore.py (hunk 15 above)
Comment (gagan3012) [reply tests/text/test_bertscore.py:40]: They are from the bert-score repo tests

**Comment by gagan3012** [comment]
How can I improve coverage I'm a bit confused

**Comment by Borda** [comment]
> How can I improve coverage I'm a bit confused

let's focus first on passing all test, then see can check coverage, but in principle, we shall have a test code covering each line of our codebase...

File: docs/source/references/functional.rst [hunk 17]
```diff
@@ -365,6 +365,10 @@ retrieval_normalized_dcg [func]
 ****
 Text
 ****
+bert_score [func]
```
Comment (SkafteNicki) [inline docs/source/references/functional.rst:368]:
```suggestion

bert_score [func]
```

File: docs/source/references/modules.rst [hunk 18]
```diff
@@ -523,6 +523,11 @@ RetrievalNormalizedDCG
 ****
 Text
 ****
+BERTScore
```
Comment (SkafteNicki) [inline docs/source/references/modules.rst:526]:
```suggestion

BERTScore
```

File: tests/text/test_bertscore.py [hunk 19]
```diff
@@ -0,0 +1,56 @@
+from typing import Any
+
+import numpy as np
+import pytest
+
+from torchmetrics.functional import bert_score
+from torchmetrics.text import BERTScore
+from torchmetrics.utilities.imports import _BERTSCORE_AVAILABLE
+
```
Comment (SkafteNicki) [inline tests/text/test_bertscore.py:9]:
```suggestion

# Examples and expected values taken from: 
# https://github.com/Tiiiger/bert_score/blob/master/tests/test_scorer.py
```

File: tests/text/test_bertscore.py [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+    "week.",
+    "But the victim's brother says he can't think of anyone who would want to hurt him, saying, \"Things were finally "
+    'going well for him."',
+]
+
+
+def _assert_list(preds: Any, refs: Any, threshold: float = 1e-8):
+    """Assert two lists are equal."""
+    assert np.allclose(preds, refs, atol=threshold, equal_nan=True)
+
```
Comment (SkafteNicki) [inline tests/text/test_bertscore.py:29]: for checking that accumulation in the modular version works
```suggestion
preds_batched = [preds[0:2], preds[2:]]
refs_batched = [target[0:2], target[2:]]
```

File: tests/text/test_bertscore.py [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+)
+@pytest.mark.skipif(not _BERTSCORE_AVAILABLE, reason="test requires bert_score")
+def test_score(preds, refs):
+    """Tests for metric."""
+    Scorer = BERTScore(model_type="roberta-large", num_layers=17, idf=False, batch_size=3)
+    Scorer.update(predictions=preds, references=refs)
+    Score = Scorer.compute()
+    _assert_list(Score["precision"], [0.9843302369117737, 0.9832239747047424, 0.9120386242866516])
+    _assert_list(Score["recall"], [0.9823839068412781, 0.9732863903045654, 0.920428991317749])
+    _assert_list(Score["f1"], [0.9833561182022095, 0.9782299995422363, 0.916214644908905])
```
Comment (SkafteNicki) [inline tests/text/test_bertscore.py:56]:
```suggestion
    _assert_list(Score["f1"], [0.9833561182022095, 0.9782299995422363, 0.916214644908905])
    

@pytest.mark.parametrize(
    "preds,refs",
    [(preds_batched, refs_batched)],
)
@pytest.mark.skipif(not _BERTSCORE_AVAILABLE, reason="test requires bert_score")
def test_accumulation(preds, refs):
    """Tests for metric works with accumulation"""
    Scorer = BERTScore(model_type="roberta-large", num_layers=17, idf=False, batch_size=3)
    for p, r in zip(preds, refs):
        Scorer.update(predictions=p, references=r)
    Score = Scorer.compute()
    _assert_list(Score["precision"], [0.9843302369117737, 0.9832239747047424, 0.9120386242866516])
    _assert_list(Score["recall"], [0.9823839068412781, 0.9732863903045654, 0.920428991317749])
    _assert_list(Score["f1"], [0.9833561182022095, 0.9782299995422363, 0.916214644908905])
```

File: torchmetrics/functional/text/bert.py [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+
+from torchmetrics.utilities.imports import _BERTSCORE_AVAILABLE
+
+if _BERTSCORE_AVAILABLE:
+    import bert_score as bert
+
+
+def bert_score(
+    predictions: List,
+    references: List,
```
Comment (SkafteNicki) [inline torchmetrics/functional/text/bert.py:24]:
```suggestion
    predictions: List[str],
    references: List[str],
```

File: torchmetrics/text/bert.py [hunk 23]
```diff
@@ -0,0 +1,125 @@
...
+        self.batch_size = batch_size
+        self.device = device
+        self.idf = idf
+        self.verbose = verbose
+        self.num_layers = num_layers
+        self.model_type = model_type
+        self.add_state("predictions", [], dist_reduce_fx="cat")
+        self.add_state("references", [], dist_reduce_fx="cat")
+
+    def update(self, predictions: List, references: List) -> None:  # type: ignore
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:97]:
```suggestion
    def update(self, predictions: List[str], references: List[str]) -> None:  # type: ignore
```

File: torchmetrics/text/bert.py [hunk 24]
```diff
@@ -0,0 +1,125 @@
...
+
+    def compute(self) -> Dict:
+        """Calculate Bertscores.
+
+        Return:
+            Dict with Bertscores.
+        """
+        return bert_score(
+            predictions=self.predictions[0],
+            references=self.references[0],
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:115]: it seems wrong to me that we only want to give the first element of the updated list to the `bert_score` function. You probably need some flattening function (see other comment) to go from a list of list to a single list of strings
```suggestion
            predictions=_flatten(self.predictions),
            references=_flatten(self.references),
```

File: torchmetrics/text/bert.py [hunk 25]
```diff
@@ -0,0 +1,125 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, List, Optional
+
+from torchmetrics.functional import bert_score
+from torchmetrics.metric import Metric
+
+
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:19]:
```suggestion
def _flatten(x: List[List[str]]) -> List[str]:
    """ converts list of list to single list of strings """
    return [e for y in x for e in y]


```

**Review by SkafteNicki — COMMENTED** [review]
more comments

File: torchmetrics/functional/text/bert.py [hunk 26]
```diff
@@ -0,0 +1,122 @@
...
+    predictions: List[str],
+    references: List[str],
+    lang: str = "en",
+    model_type: Optional[str] = None,
+    num_layers: int = None,
+    verbose: bool = False,
+    idf: bool = False,
+    device: Optional[str] = None,
+    batch_size: int = 64,
+    nthreads: int = 4,
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:32]:
```suggestion
    num_threads: int = 4,
```

File: torchmetrics/functional/text/bert.py [hunk 27]
```diff
@@ -0,0 +1,122 @@
...
+                  specified when `rescale_with_baseline` is True.
+        - `rescale_with_baseline` (bool): rescale bertscore with pre-computed baseline
+        - `baseline_path` (str): customized baseline file
+
+    Returns:
+        - `(P, R, F)`: each is of shape (N); N = number of input
+                      candidate reference pairs. if returning hashcode, the
+                      output will be ((P, R, F), hashcode). If a candidate have
+                      multiple references, the returned score of this candidate is
+                      the *best* score among all references.
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:67]: Pls use Napoleon style, see other docs

File: torchmetrics/functional/text/bert.py [hunk 28]
```diff
@@ -0,0 +1,122 @@
...
+        nthreads=nthreads,
+        all_layers=all_layers,
+        idf=idf,
+        device=device,
+        lang=lang,
+        rescale_with_baseline=rescale_with_baseline,
+        baseline_path=baseline_path,
+    )
+
+    (P, R, F) = cached_bertscorer.score(
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:110]:
```suggestion
    P, R, F = cached_bertscorer.score(
```

File: torchmetrics/functional/text/bert.py [hunk 29]
```diff
@@ -0,0 +1,122 @@
...
+    hashcode = bert.get_hash(
+        model=model_type,
+        num_layers=num_layers,
+        idf=idf,
+        rescale_with_baseline=rescale_with_baseline,
+        use_custom_baseline=baseline_path is not None,
+        use_fast_tokenizer=True,
+    )
+
+    cached_bertscorer = bert.BERTScorer(
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:97]:
```suggestion
    cached_bertscorer = BERTScorer(
```

File: torchmetrics/functional/text/bert.py [hunk 30]
```diff
@@ -0,0 +1,122 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, List, Optional
+
+from torchmetrics.utilities.imports import _BERTSCORE_AVAILABLE
+
+if _BERTSCORE_AVAILABLE:
+    import bert_score as bert
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:19]:
```suggestion
    from bert_score import BERTScorer
```

File: torchmetrics/text/bert.py [hunk 31]
```diff
@@ -0,0 +1,130 @@
...
+    return [e for y in x for e in y]
+
+
+class BERTScore(Metric):
+    """BERTScore leverages the pre-trained contextual embeddings from BERT and matches words in candidate and
+    reference sentences by cosine similarity. It has been shown to correlate with human judgment on sentence-level
+    and system-level evaluation. Moreover, BERTScore computes precision, recall, and F1 measure, which can be
+    useful for evaluating different language generation tasks.
+
+    Args:
```
Comment (Borda) [inline torchmetrics/text/bert.py:31]: again, pls use the same style as all TM codebase :]

**Review by Borda — CHANGES_REQUESTED** [review]

File: torchmetrics/functional/text/bert.py (hunk 30 above)
Comment (gagan3012) [reply torchmetrics/functional/text/bert.py:19]: Fixed

File: torchmetrics/functional/text/bert.py (hunk 29 above)
Comment (gagan3012) [reply torchmetrics/functional/text/bert.py:87]: Fixed

File: torchmetrics/functional/text/bert.py (hunk 26 above)
Comment (gagan3012) [reply torchmetrics/functional/text/bert.py]: Fixed

File: torchmetrics/functional/text/bert.py [hunk 32]
```diff
@@ -0,0 +1,112 @@
...
+
+    Returns:
+        (Dict): containing: Precision, Recall, F1 score, Hashcode of the library
+
+    Example:
+        >>> predictions = ["hello there", "general kenobi"]
+        >>> references = ["hello there", "general kenobi"]
+        >>> results = bert_score(predictions=predictions, references=references, lang="en")
+        >>> print([round(v, 2) for v in results["f1"]])
+        [1.0, 1.0]
```
Comment (SkafteNicki) [inline torchmetrics/functional/text/bert.py:64]: any reason we are not showing the hole `results` dict?

File: torchmetrics/text/bert.py [hunk 33]
```diff
@@ -0,0 +1,133 @@
...
+        self.model_type = model_type
+        self.add_state("predictions", [], dist_reduce_fx="cat")
+        self.add_state("references", [], dist_reduce_fx="cat")
+
+    def update(self, predictions: List[str], references: List[str]) -> None:  # type: ignore
+        """Store predictions/references for computing BERT scores.
+
+        Args:
+            predictions: List of predicted sentences
+            references: List of refernces
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:109]:
```suggestion
            references: List of references
```

File: torchmetrics/text/bert.py [hunk 34]
```diff
@@ -0,0 +1,133 @@
...
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step. default: False
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the allgather
+
+    Returns:
+        (Dict): containing: Precision, Recall, F1 score, Hashcode of the library
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:56]:
```suggestion
        Dict containing the keys `precision`, `recall`, `f1` and `hashcode` with 
        corresponding values
```

File: torchmetrics/text/bert.py [hunk 35]
```diff
@@ -0,0 +1,133 @@
...
+
+    Example:
+        >>> predictions = ["hello there", "general kenobi"]
+        >>> references = ["hello there", "general kenobi"]
+        >>> results = bert_score(predictions=predictions, references=references, lang="en")
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:61]:
```suggestion
        >>> bert_score = BertScore(lang="en")
        >>> results = bert_score(predictions=predictions, references=references)
```

File: torchmetrics/functional/text/bert.py [hunk 36]
```diff
@@ -0,0 +1,112 @@
...
+    device: Optional[str] = None,
+    batch_size: int = 64,
+    num_threads: int = 4,
+    all_layers: bool = False,
+    rescale_with_baseline: bool = False,
+    baseline_path: Optional[str] = None,
+) -> Dict:
+    """BERTScore leverages the pre-trained contextual embeddings from BERT and matches words in candidate and
+    reference sentences by cosine similarity. It has been shown to correlate with human judgment on sentence-level
+    and system-level evaluation. Moreover, BERTScore computes precision, recall, and F1 measure, which can be
```
[the remark this answers is not in the document]
Comment (gagan3012) [reply torchmetrics/functional/text/bert.py:39]: This is the relevant paper: https://arxiv.org/abs/1904.09675

File: torchmetrics/functional/text/bert.py (hunk 32 above)
Comment (gagan3012) [reply torchmetrics/functional/text/bert.py:65]: Updated to have the entire result

**Review by Borda — COMMENTED** [review]
later we shall come with our own BERT implementation as the `bert_score` require heavy Transformers dependency

**Review by Borda — APPROVED** [review]

File: torchmetrics/text/bert.py [hunk 37]
```diff
@@ -0,0 +1,139 @@
...
+    return [e for y in x for e in y]
+
+
+class BERTScore(Metric):
+    """`BERTScore <https://arxiv.org/abs/1904.09675>`_ leverages the pre-trained contextual embeddings from BERT
+    and matches words in candidate and reference sentences by cosine similarity. It has been shown to correlate
+    with human judgment on sentence-level and system-level evaluation. Moreover, BERTScore computes precision,
+    recall, and F1 measure, which can be useful for evaluating different language generation tasks.
+
+    <https://arxiv.org/abs/1904.09675>
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:31]:
```suggestion
```

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**